x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -633,6 +633,14 @@
         Setting.Property.NodeScope
     );
 
+    /**
+     * Each model deployment results in one or more entries in the cluster state
+     * for the model allocations. In order to prevent the cluster state from
+     * potentially growing uncontrollably we impose a limit on the number of
+     * trained model deployments.
+     */
+    public static final int MAX_TRAINED_MODEL_DEPLOYMENTS = 200;
```

Comment:
My only question about this would be whether it's better to start with a lower limit, say 100, then have room to increase it in a future version if there are complaints that 100 is too low.

The danger with starting off at the real limit is that if we find some reason why we need to reduce it in the future then that would be a breaking change, whereas increasing it is just an enhancement that won't detrimentally impact anyone.

Reply:
Agreed. Changed it to `100`.